src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt

```diff
@@ -214,9 +214,9 @@ class ActivitiesService(
 
   private fun toNomisActivityCode(activityDescription: String, scheduleDescription: String): String {
     var code = "SAA-$activityDescription"
-      .let { it.substring(0, min(it.length, 20)).trim() }
-    if (code.length < 19) {
-      code += "-${scheduleDescription.let { it.substring(0, min(it.length, 19 - code.length)) }}"
+      .let { it.substring(0, min(it.length, 12)).trim() }
```

Comment:
can we get duplicates and if we do what do we do about it?

Reply:
Hmmm there's no constraint so I thought it wouldn't matter. However the UI has validation to prevent duplicates so I guess I need to emulate that 🤔